### PR TITLE
Storybook: Add WritingModeControl story

### PR DESCRIPTION
**Pull request by Sukhendu2002**
Part of https://github.com/WordPress/gutenberg/issues/67165

## What?
This PR will add stories for WritingModeControl component in the Storybook.

## Testing Instructions
1. Run npm run storybook:dev
2. Open the storybook on http://localhost:50240/
3. Check the WritingModeControl stories.

## Screenshots or screencast

https://github.com/user-attachments/assets/eabbdd34-221a-467c-a5cc-c5a5838b0957

### CODE REVIEW

File: packages/block-editor/src/components/writing-mode-control/stories/index.story.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			control: { type: 'select' },
+			options: ( () => {
+				const modes = [
+					'horizontal-tb',
+					isRTL() ? 'vertical-lr' : 'vertical-rl',
+				];
+				return modes;
+			} )(),
+			description: 'Currently selected writing mode.',
+		},
```
Comment (t-hamano) [inline packages/block-editor/src/components/writing-mode-control/stories/index.story.js]:
```suggestion
		value: {
			control: { type: null },
			description: 'Currently selected writing mode.',
		},
```

File: packages/block-editor/src/components/writing-mode-control/stories/index.story.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		className: {
+			control: 'text',
+			description: 'Class name to add to the control.',
+		},
+		onChange: {
+			action: 'onChange',
```
Comment (t-hamano) [inline packages/block-editor/src/components/writing-mode-control/stories/index.story.js:32]:
```suggestion
			action: 'onChange',
			control: { type: null },
```

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

In order to move this PR forward, I would like to suggest two points that I think are particularly important.

1 - Only one story is needed: I think the only difference between the Default and Vertical stories is the default value. In this case, I think an additional story (Vertical) is unnecessary.

2 - Don't make the `value` prop controllable: In the current implementation, nothing happens when you press the component's button after changing the value prop.

https://github.com/user-attachments/assets/72b6ab83-0e75-43eb-9500-ab00f5d0e967

How about implementing it in control mode as follows?

```jsx
export const Default = {
	render: function Template( { onChange, ...args } ) {
		const [ value, setValue ] = useState();
		return (
			<WritingModeControl
				{ ...args }
				onChange={ ( ...changeArgs ) => {
					onChange( ...changeArgs );
					setValue( ...changeArgs );
				} }
				value={ value }
			/>
		);
	},
};
```

**Comment by Sukhendu2002** [comment]
Hi @t-hamano, Thanks for the detailed feedback!

I’ve updated the story:
- The Default and Vertical stories have been combined into a single story, as the difference was only the default value.
- The component now manages the value internally using useState. 

Please let me know if there’s anything else that needs adjustment!

File: packages/block-editor/src/components/writing-mode-control/stories/index.story.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import WritingModeControl from '../';
+
+/**
+ * WritingModeControl component allows selecting writing mode.
+ */
```
Comment (t-hamano) [inline packages/block-editor/src/components/writing-mode-control/stories/index.story.js]: The description is already defined [here](https://github.com/WordPress/gutenberg/pull/67343/files#diff-c836fbdea3b54e55e637d06ffc45758c46f13ac4b2d99a38bd2ca0fdaf5a6bedR22), so this comment is unnecessary.

File: packages/block-editor/src/components/writing-mode-control/stories/index.story.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	},
+};
+
+export default meta;
+
+export const Default = {
+	render: function Template( { onChange, ...args } ) {
+		const [ value, setValue ] = useState(
+			isRTL() ? 'vertical-lr' : 'vertical-rl'
+		);
```
Comment (t-hamano) [inline packages/block-editor/src/components/writing-mode-control/stories/index.story.js]:
```suggestion
		const [ value, setValue ] = useState();
```

I don't think there's a default value needed.

**Review by t-hamano — COMMENTED** [review]
Thanks for the update! Overall it looks good, but I'd like to suggest two additional illustrations at the end.

**Comment by Sukhendu2002** [comment]
Hi @t-hamano, Thanks for the feedback! I’ve addressed both points.

**Review by t-hamano — APPROVED** [review]
LGTM! Works as expected 👍


https://github.com/user-attachments/assets/a451a462-9e4c-4316-b85d-06eb8494c4db

**Pull request merged**